declarations_site/chatbot/views.py

```diff
@@ -96,27 +255,133 @@ def search_reply(data):
+                qs={'q': data['text'], 'deepsearch': deepsearch})
+            if notify_id:
+                next_page = "нові {}/{}".format(notify_id, skip + count)
+            else:
+                next_page = "{} /{}".format(data['text'], skip + count)
+            att = {
+                "contentType": "application/vnd.microsoft.card.hero",
+                "content": {
+                    "title": "Більше декларацій",
+                    "buttons": [
+                        {
+                            "type": "imBack",
+                            "title": "Наступні {} декларацій".format(count),
+                            "value": next_page,
+                        },
+                        {
+                            "type": "openUrl",
+                            "title": "Перейти на сайт",
+                            "value": url
+                        }
+                    ]
+                }
+            }
+
+            attachments.append(att)
+
+        if not find_subscription2(data, data['text']):
+            att = {
+                "contentType": "application/vnd.microsoft.card.hero",
+                "content": {
+                    "title": "Підписатись на запит \"{}\"".format(data['text']),
+                    "text": "Отирмуйте оновлення по цьому запиту миттєво в чат",
+                    "buttons": [
+                        {
+                            "type": "imBack",
+                            "title": "Підписатись",
+                            "value": "підписатись {}".format(data['text'])
+                        }
+                    ]
+                }
+            }
+        else:
+            att = {
+                "contentType": "application/vnd.microsoft.card.hero",
+                "content": {
+                    "title": "Ви підписані на оновлення",
+                    "text": "Щоб відписатись він наступних повідомлень по цьому запиту оберіть:",
+                    "buttons": [
+                        {
+                            "type": "imBack",
+                            "title": "Відписатись",
+                            "value": "відписатись {}".format(data['text'])
+                        }
+                    ]
+                }
+            }
+
+        attachments.append(att)
+
+    return attachments
+
+
+def search_reply(data):
+    if not data.get('text') or len(data['text']) > 100:
+        return chat_response(data, 'Не зрозумів, уточніть запит.')
+
+    text = data['text'].strip(' .,;:!-()\n').lower()
+
+    for r, message in QUICK_ANSWERS:
+        if r.match(text):
+            if isinstance(message, (list, tuple, set)):
+                message = choice(message)
+            return chat_response(data, message)
+
+    for r, handler in CHATBOT_COMMANDS:
+        if r.match(text):
+            return handler(data)
+
+    # deepsearch switch
+    deepsearch = False
+    if ' /' in text:
+        for deep in (' /deep', ' /всюди'):
```

Comment:
Now it's only for testing purposes, if you think its not difficult to explain, it may be added to the docs later.